nacc_form_validator/nacc_validator.py

```diff
@@ -669,20 +669,52 @@ def _validate_temporalrules(self, temporalrules: List[Mapping], field: str,
                             'type': 'dict',
                             'required': True,
                             'empty': False
+                        },
+                        'ignore_empty': {
+                            'schema': {
+                                'oneof': [
+                                    {'type': 'string', 'empty': False},
+                                    {
+                                        'type': 'list',
+                                        'minlength': 1,
+                                        'schema': {
+                                            'type': "string"
+                                        }
+                                    }
+                                ]
+                            },
+                            "required": False
                         }
                     }
                 }
             }
         """
-        prev_ins = self.get_previous_record()
-
-        # If temporal rules are defined, a previous vist must exist
-        if prev_ins is None:
-            self._error(field, ErrorDefs.NO_PREV_VISIT)
-            return
-
         rule_no = -1
         for temporalrule in temporalrules:
+            ignore_empty_fields = temporalrule.get(SchemaDefs.IGNORE_EMPTY,
+                                                   None)
+            prev_ins = self.get_previous_record(
+                field=ignore_empty_fields,
+                ignore_empty=True if ignore_empty_fields else False)
+
+            # if not found, pass through validation
+            if not prev_ins:
+                continue
```

Comment:
Only skip validation for ignore_empty=True, for the regular case prev_ins must exist, otherwise something is wrong, raise NO_PREV_VISIT error in that case.